Classes/Eel/SourceUriHelper.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Networkteam\ImageProxy\Eel;
+
+/***************************************************************
+ *  (c) 2022 networkteam GmbH - all rights reserved
+ ***************************************************************/
+
+use Neos\Eel\ProtectedContextAwareInterface;
+use Neos\Flow\Annotations as Flow;
+use Neos\Media\Domain\Model\Asset;
+use Networkteam\ImageProxy\Aspects\ThumbnailAspect;
+
+class SourceUriHelper implements ProtectedContextAwareInterface
+{
+    /**
+     * @var ThumbnailAspect
+     * @Flow\Inject
+     */
+    protected $thumbnailAspect;
+
+    /**
+     * @param $asset
+     * @return string The source uri of the asset
+     */
+    public function sourceUri(Asset $asset)
+    {
+        return $this->thumbnailAspect->getSourceUri($asset);
```

Comment:
I'd swap this and put the logic for building the S3 URI in the helper and use it in the aspec.
It would be nicer for the aspect to depend on a helper (which is pretty "pure") than the other way around, since aspects are somewhat special and it's rather unusual to inject an aspect just for re-using a method.